File: ParadigmaComp/15Questoes/python/questao12.py
#Sobrecarga de Operadores (Python)
class Produto:
    def __init__(self, nome, preco):
        self.nome = nome
        self.preco = preco

    def __add__(self, outro):
        return self.preco + outro.preco

# Exemplo de uso
produto1 = Produto("Teclado", 100)
produto2 = Produto("Mouse", 50)

print(produto1 + produto2)  # Soma dos preços dos produtos

File: ParadigmaComp/15Questoes/golang/questao12.go
package main
//Métodos de Conveniência (Go)
import "fmt"

// Estrutura Produto
type Produto struct {
	Nome  string
	Preco float64
}

// Método que soma os preços de dois produtos
func SomarProdutos(p1, p2 Produto) float64 {
	return p1.Preco + p2.Preco
}

func main() {
	produto1 := Produto{Nome: "Teclado", Preco: 100}
	produto2 := Produto{Nome: "Mouse", Preco: 50}

	fmt.Println(SomarProdutos(produto1, produto2))  // Soma dos preços dos produtos
}
